### PR TITLE
[air/output] Improve passed time display

### LINKED ISSUE #34922: [AIR output] Explain what the time refers to and render time in a more readable format.
### What happened + What you expected to happen

1. When I ran `tune_torch_benchmark.py` with the new air output, the time info is sometimes ambiguous.
```
# current
Current time: 2023-03-28 19:11:40 (running for 00:01:39.40)

# suggestion
Current time: 2023-03-28 19:11:40. Total running time: 1min, 39.40s
```

2. In general, can we make the time info more understandable?
Training time of this iteration == time_this_iter 
Total running time == time_total_s at the end of last iteration
Total training time == sum of all iterations' time_this_iter?

Something like this for a train-only user?
```
# Current
Training finished iter 4 at 2023-03-28 10:09:42 (running for 00:00:44.94)
loss: 1.8674473033589163
time_this_iter_s: 8.962362051010132
time_total_s: 39.676079988479614
training_iteration: 4

Training (4 iters) finished at 2023-03-28 10:09:44 (running for 00:00:46.97)
loss: 1.8674473033589163
time_this_iter_s: 8.962362051010132
time_total_s: 39.676079988479614
training_iteration: 4

Last result: {'loss': 1.8674473033589163, 'timestamp': 1680023382, 'time_this_iter_s': 8.962362051010132, 'done': True, 'training_iteration': 4, 'trial_id': '39f11_00000', 'date': '2023-03-28_10-09-42', 'time_total_s': 39.676079988479614, 'pid': 73070, 'hostname': 'ip-10-0-23-28', 'node_ip': '10.0.23.28', 'config': {'train_loop_config': {'lr': 0.001, 'batch_size': 64, 'epochs': 4}}, 'time_since_restore': 39.676079988479614, 'iterations_since_restore': 4, 'experiment_tag': '0'}



# Suggestion
Training finished iteration 4 at 2023-03-28 10:09:42. Training time of this iteration: 8.96s
-------------------------
loss             2.24726
-------------------------

Training finished (4 iterations) at 2023-02-24 12:35:39. Total training time: 39.68s. Total running time: 46.97s
Last reported results
-------------------------
loss             2.24726
-------------------------
```

### Versions / Dependencies

nightly

### Reproduction script

tune_torch_benchmark.py

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Includes #34788 

This changes the formatting of the total running time like this:

```
Trial easy_objective_b12ae_00003 completed training after 5 iterations at 2023-05-02 14:46:11. Total running time: 4d 3hr 20min, 2s

Trial easy_objective_d3d85_00008 completed training after 5 iterations at 2023-05-02 14:47:09. Total running time: 4d 0hr 20min 2s

Trial easy_objective_cbd93_00007 finished iteration 5 at 2023-05-02 14:46:56. Total running time: 20min 2s

Trial easy_objective_14e6b_00001 completed training after 5 iterations at 2023-05-02 14:48:58. Total running time: 1hr 0min 2s
```

It does not turn metrics (e.g. `time_this_iter` or `total_time_s`) into written sentences, as this will make it harder to identify these metrics for configuration purposes.

## Related issue number

Closes #34922

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
LGTM. Thanks!

**Comment by richardliaw** [comment]
FYI i'm not really a fan of the human-readable time, but that's ok for now. let's leave it to the users to decide.

File: python/ray/tune/experimental/output.py [hunk 1]
```diff
@@ -110,22 +110,28 @@ def _get_time_str(start_time: float, current_time: float) -> Tuple[str, str]:
...
+        running_for_str = f"{days:d}d "
     else:
         running_for_str = ""
 
-    running_for_str += f"{hours:02.0f}:{minutes:02.0f}:{seconds:05.2f}"
+    if hours > 0 or days > 0:
+        running_for_str += f"{hours:d}hr "
+
+    if minutes > 0 or hours > 0 or days > 0:
+        running_for_str += f"{minutes:d}min "
```
Comment (gjoliver) [inline python/ray/tune/experimental/output.py:132]: would it be simpler to just check whether running_for_str already has some content? like
```
running_for_str = ""
if days > 0:
    running_for_str += f"{days:d}d "
if hours > 0 or running_for_str:
    running_for_str += f"{hours:d}hr "
if minutes > 0 or running_for_str:
    running_for_str += f"{minutes:d}min "
running_for_str += f"{seconds:d}s"
```

**Review by gjoliver — COMMENTED** [review]
nice. just a couple of nits.

File: python/ray/tune/experimental/output.py (hunk 1 above)
Comment (krfricke) [reply python/ray/tune/experimental/output.py:131]: Sure, good idea. Updated!

**Review by gjoliver — APPROVED** [review]

**Pull request merged**